src/main/java/com/cburch/logisim/fpga/hdlgenerator/ToplevelHDLGeneratorFactory.java

```diff
@@ -120,7 +119,23 @@ public ToplevelHDLGeneratorFactory(long fpgaClock, double tickClock, Circuit top
           ledArray.getNrOfRows(),
           ledArray.getNrOfColumns(),
           myLedArrays.indexOf(ledArray)));
+      final var ports = LedArrayGenericHDLGeneratorFactory.getExternalSignals(
+          ledArray.getArrayDriveMode(),
+          ledArray.getNrOfRows(),
+          ledArray.getNrOfColumns(),
+          myLedArrays.indexOf(ledArray));
+      for (var port : ports.keySet())
```

Comment:
`final var port` please (+ a few `for`s below). In general, please always make variables `final` unless needed otherwise.

Reply:
done